--- src/TerrariaNetCore/Terraria/GameContent/UI/BigProgressBar/BigProgressBarSystem.cs
+++ src/tModLoader/Terraria/GameContent/UI/BigProgressBar/BigProgressBarSystem.cs
@@ -2,10 +_,11 @@
 using Microsoft.Xna.Framework;
 using Microsoft.Xna.Framework.Graphics;
 using Terraria.IO;
+using Terraria.ModLoader;
 
 namespace Terraria.GameContent.UI.BigProgressBar;
 
-public class BigProgressBarSystem
+public partial class BigProgressBarSystem
 {
 	private IBigProgressBar _currentBar;
 	private CommonBossBigProgressBar _bossBar = new CommonBossBigProgressBar();
@@ -62,17 +_,31 @@
 
 	public void Update()
 	{
+		if (BossBarLoader.CurrentStyle.PreventUpdate)
+			goto endUpdate;
+
 		if (_currentBar == null)
 			TryFindingNPCToTrack();
 
 		if (_currentBar != null && !_currentBar.ValidateAndCollectNecessaryInfo(ref _info))
 			_currentBar = null;
+
+		endUpdate:
+		BossBarLoader.CurrentStyle.Update(_currentBar, ref _info);
 	}
 
 	public void Draw(SpriteBatch spriteBatch)
 	{
+		if (BossBarLoader.CurrentStyle.PreventDraw)
+			goto endDraw;
+
-		if (_currentBar != null)
+		if (_currentBar != null) {
+			BossBarLoader.drawingInfo = _info;
 			_currentBar.Draw(ref _info, spriteBatch);
+		}
+
+		endDraw:
+		BossBarLoader.CurrentStyle.Draw(spriteBatch, _currentBar, _info);
 	}
 
 	private void TryFindingNPCToTrack()
@@ -80,7 +_,7 @@
 		Rectangle value = new Rectangle((int)Main.screenPosition.X, (int)Main.screenPosition.Y, Main.screenWidth, Main.screenHeight);
 		value.Inflate(5000, 5000);
 		float num = float.PositiveInfinity;
-		for (int i = 0; i < 200; i++) {
+		for (int i = 0; i < Main.maxNPCs; i++) {
 			NPC nPC = Main.npc[i];
 			if (nPC.active && nPC.Hitbox.Intersects(value)) {
 				float num2 = nPC.Distance(Main.LocalPlayer.Center);
@@ -92,7 +_,7 @@
 
 	public bool TryTracking(int npcIndex)
 	{
-		if (npcIndex < 0 || npcIndex > 200)
+		if (npcIndex < 0 || npcIndex > Main.maxNPCs)
 			return false;
 
 		NPC nPC = Main.npc[npcIndex];
@@ -103,14 +_,19 @@
 		bigProgressBarInfo.npcIndexToAimAt = npcIndex;
 		BigProgressBarInfo info = bigProgressBarInfo;
 		IBigProgressBar bigProgressBar = _bossBar;
+
+		//BossBars assigned by mods take priority
+		if (nPC.BossBar != null) {
+			bigProgressBar = nPC.BossBar;
+		}
-		if (_bossBarsByNpcNetId.TryGetValue(nPC.netID, out var value))
+		else if (_bossBarsByNpcNetId.TryGetValue(nPC.netID, out var value))
 			bigProgressBar = value;
 
+		info.showText = true; // Moved so that mods can set it to false
 		if (!bigProgressBar.ValidateAndCollectNecessaryInfo(ref info))
 			return false;
 
 		_currentBar = bigProgressBar;
-		info.showText = true;
 		_info = info;
 		return true;
 	}
